src/attacks/fgsm/fgsm.jl

```diff
@@ -7,11 +7,19 @@ using Flux: onehotbatch, onecold
 
 # White-box Fast Gradient Sign Method (FGSM) attack by Goodfellow et al. (arxiv.org/abs/1412.6572)
 # Code adapted from (github.com/jaypmorgan/Adversarial.jl)
-function FGSM(model, x, y; loss=cross_entropy_loss, ϵ = 0.3, min_label=0, max_label=9, clamp_range = (0, 1))
-    w, h, c = size(x)
-    x_adv = deepcopy(x)
-    x_adv = reshape(x_adv, w, h, c, 1)
-    grads = gradient(x_adv -> loss(model(x_adv), y; min_label=min_label, max_label=max_label)[1], x_adv)[1]
+function FGSM(
+    model,
+    x,
+    y;
+    loss = cross_entropy_loss,
+    ϵ = 0.3,
+    clamp_range = (0, 1),
+)
+    x_adv = reshape(x, size(x)..., 1)       # why is this necessary @rithik83?
```

Comment:
why do we need this reshape?

Reply:
The model I used to test, accepted inputs of the shape (w, h, c, batch_size). This function (and all attacks) have an input of a single image/datapoint etc (shape: (w, h, c)). so to make it compatible with the model to be able to test and iterate, I reshaped.

How do you suggest we do it? passing inputs as a batch probably is better long-term (speedups and what not) but then we may have to alter the attack algorithms to be able to deal with it. 
or for now, pass an input as (w, h, c, 1) itself so it doesn't have to keep being reshaped

Reply:
I think we probably want to use broadcasting for that instead